### DISCUSSION TITLE
feat: New config option staticRepoConfig

### DISCUSSION THREAD

**Discussion by nabeelsaabna** [renovatebot/renovate#33280] [Suggest an Idea]
### Tell us more.

Mend needs a new config option called `RENOVATE_STATIC_REPO_CONFIG`
This config option is part of the “self-hoted config” (can’t be used from the repository `renovate.json`)
Renovate will evaluate this new config for each repository immediately before reading/processing the repository config 

Current config order:

1. Default config
2. Global config
3. File config
4. Environment config
5. CLI config
6. Inherited config
7. Resolved presets referenced in config
8. Repository config

The new option should be applied and merged after `#6` and before `#7`

better naming options than `staticRepoConfig` are welcomed, preferably a name that is not in use already to reduce confusion.

**Comment by RahulGautamSingh** [comment]
What will be the use case for this option? To make it easier to tweak repo config say for testing..

**Reply by rarkins** [reply]
Mend has an alternative config file ".whitesource" which should be treated as repo config. It has a complicated logic to resolve and we don't want to add that to Renovate, so instead we resolve it before calling Renovate and then tell Renovate "here is your repo config".
